src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -371,6 +371,9 @@ private void insertValidationEventListeners(Injector injector) {
      * @throws RuntimeException if there are missing changesets or a LiqubaseException
      */
     protected void checkDbChangelog() {
+        if (!config.getBoolean(HALT_ON_LIQUIBASE_DESYNC)) {
+            return;
+        }
```

Comment:
I think that we should still log out the missing changesets so that it is clear that there are missing changesets, but then have a log message that states that we will omit these changes due to the configuration. I think if we hide the missing changesets from the logs it will be easy for people to add a false to the halt config and forget about it. I'm open to opinions on this though!

We could put the "Please update the database.", "Aborting Candlepin initialization ...", and the RuntimeException in a conditional for when the HALT_ON_LIQUIBASE_DESYNC is true so that it is displayed only when CP should halt.

Reply:
The check for missing changesets on startup was added for a specific use case: upstream katello/foreman/candlepin users.
When it comes to Satellite customers, they don't actually need this check because the satellite installer tooling takes care of database upgrades (it's just a nice-to-have insurance policy that they'll probably never need).
And when it comes to Hosted, they always check if/what the changesets that need to be run are before an upgrade, so they don't have a need for it (that's why they asked for this config to disable it entirely).

So tl;dr, I don't think any big issue would be caused if someone turned this check off and forgot about it, and for hosted to be able to turn the check off entirely also gives a startup speed boost (since the liquibase db tables arent checked).

Reply:
I see! Thanks for the context around the check. I'm fine with not logging out the missing changesets since they aren't crucial to any of the consumers in this case!

Reply:
This is a case where if we wanted the logging (which is a fair ask), it'd be separate configurations: one for whether or not to run the check at all including all the logs, and one to actually allow CP to continue running in such a state.

Given the initial implementation of this PR, that's where I expected it to go, but it's not strictly required for the RFE in question, and I'm not going to hold up the PR for it to be added back in.